app/src/main/java/com/example/websocketserver/MySocketServer.java

```diff
@@ -28,7 +28,7 @@ public void onOpen(WebSocket conn, ClientHandshake handshake) {
 
     @Override
     public void onClose(WebSocket conn, int code, String reason, boolean remote) {
-
+  mSocket.close();
     }
 
```

Comment:
Can we just keep this file and remove everything else that may be unnecessary?  I believe this is the server core? May I suggest that you clean the code here a bit. Take care of indentations etc.